umpost/um2netcdf.py

```diff
@@ -603,22 +594,58 @@ def check_pressure_warnings(need_heaviside_uv, heaviside_uv, need_heaviside_t, h
               "These fields will be skipped")
 
 
+def filtered_cubes(cubes, include=None, exclude=None):
+    """
+    Generator filters & emits cubes by include or exclude lists.
+
+    Include & exclude args are mutually exclusive. If neither include or exclude
+    are specified, the generator yields the full cube list.
+
+    Parameters
+    ----------
+    cubes : Sequence of Iris Cube objects
+    include: Sequence of item_code (int) to include (discarding all others)
+    exclude: Sequence of item_code (int) to exclude (keeping all other cubes)
+    """
+    if include and exclude:
+        msg = "Include and exclude lists are mutually exclusive"
+        raise ValueError(msg)
+
```

Comment:
I am not sure abt this condition. Are you checking if the include and exclude list should not have anything in common? In that case convert list to set and compare with bitwise AND

Reply:
This is from the original utility. Users can select variables/cubes by name. The original CLI specifies that a user can supply a list of var names to include XOR a list of names to exclude. As far as I can tell, the original code tried to avoid items in both lists.

I'm going to leave the interface as is as the main goal of step 1 is adding unit tests/restructuring.